### PR TITLE
AP-49 Ensure user is requesting his hub

### PR DESCRIPTION
**Pull request by Alaggaerion**
Added if statement to ensure user is requesting his corresponding hub

### CODE REVIEW

File: mobile_app_connector/controllers/mobile_app_controller.py [hunk 1]
```diff
@@ -73,9 +73,13 @@ def get_hub_messages(self, partner_id, **parameters):
         """
         hub_obj = request.env['mobile.app.hub'].sudo()
         if partner_id:
-            # This will ensure the user is logged in
-            request.session.check_security()
-            hub_obj = hub_obj.sudo(request.session.uid)
+            # Check if requested url correspond to the current user
+            if partner_id == request.env.user.partner_id.id:
```
Comment (ecino) [inline mobile_app_connector/controllers/mobile_app_controller.py:77]: Did you test it? Are you sure `request.env.user` is set at that point? I'm not sure that the variable exists here but I may be wrong.

Comment (Alaggaerion) [reply mobile_app_connector/controllers/mobile_app_controller.py:77]: Yes it works, I already tested it and I just checked again

Comment (ecino) [reply mobile_app_connector/controllers/mobile_app_controller.py:77]: Ok, then it looks good. However it's breaking the automated tests in `test_mobile_app_http`. Can you have a look at it? You can see the errors in the Travis build.

File: mobile_app_connector/tests/test_mobile_app_http.py [hunk 2]
```diff
@@ -55,11 +55,11 @@ def test_hub_authentication(self, oauth_patch):
         json_data = simplejson.loads(response.read())
         self.assertIn('Messages', json_data)
         # Private messages without login should fail
-        response = self.url_open(url + '1')
+        response = self.url_open(url + '3')
```
Comment (ecino) [inline mobile_app_connector/tests/test_mobile_app_http.py]: `self.env.ref('base.partner_root').id`

**Review by ecino — APPROVED** [review]

**Pull request merged**